pkg/smokescreen/smokescreen.go

```diff
@@ -541,10 +541,13 @@ func BuildProxy(config *Config) *goproxy.ProxyHttpServer {
 	proxy.OnResponse().DoFunc(func(resp *http.Response, pctx *goproxy.ProxyCtx) *http.Response {
 		sctx := pctx.UserData.(*smokescreenContext)
 
-		if resp != nil && resp.Header.Get(errorHeader) != "" {
-			if pctx.Error == nil && sctx.decision.allow {
+		if resp != nil && pctx.Error == nil && sctx.decision.allow {
```

Comment:
I think we actually want to keep these conditionals the same, or hoist up the error header check outside of this `if`. 

We're trying to delete any response headers that we didn't set (e.g. the upstream server could set a fake `X-Smokescreen-Error` header), but I think we'll miss that deletion with this change.

Reply:
I'm not sure I see how the deletion logic changes? We're replacing 

```
if a && b {
  if c && d {
    <delete>
  }
}
```

with

```
if a && c && d {
  if b {
    <delete>
  }
}
```

Either way, all four conditionals have to be true for the header to be deleted

Reply:
oh whoops you're right! it's too early 🥲

Reply:
:awake-but-at-what-cost: